src/test/java/io/numaproj/numaflow/function/FunctionServerTest.java

```diff
@@ -65,7 +65,7 @@ public void setUp() throws Exception {
         server
                 .registerMapper(new MapFunc(testMapFn))
                 .registerMapperT(new MapTFunc(testMapTFn))
-                .registerReducer(ReduceTestFn.class)
+                .registerReducer(new ReduceTestFactory())
```

Comment:
same as above, registerReducer(new ReduceTestFactory().createReducer())

Reply:
same thing, we can't use one single reducer for all the keys

Reply:
why can't we? are we gonna have `createReducerByKey(String key)`?